extensions/amp-story-subscriptions/0.1/test/test-amp-story-subscriptions.js

```diff
@@ -25,51 +28,153 @@ describes.realWin(
+            'actions': {
+              'login': 'https://scenic-2017.appspot.com/signin?rid=READER_ID',
+              'subscribe': 'https://scenic-2017.appspot.com/subscribe',
+            },
+          },
+          {
+            'serviceId': 'subscribe.google.com',
+          },
+        ],
+      };
+      subscriptionServiceConfigEl.innerHTML = JSON.stringify(platformConfig);
+      doc.head.appendChild(subscriptionServiceConfigEl);
+
+      // Specify page config
+      const pageConfigEl = <script type="application/ld+json"></script>;
+      const pageConfig = {
+        '@context': 'http://schema.org',
+        '@type': 'NewsArticle',
+        'isAccessibleForFree': 'False',
+        'isPartOf': {
+          '@type': ['CreativeWork', 'Product'],
+          'name': 'Product A',
+          'productID': 'scenic-2017.appspot.com:news',
+        },
+      };
+      pageConfigEl.innerHTML = JSON.stringify(pageConfig);
+      doc.head.appendChild(pageConfigEl);
+
+      subscriptionService = new SubscriptionService(env.ampdoc);
+      subscriptionService.start();
+      env.sandbox
+        .stub(Services, 'subscriptionsServiceForDoc')
+        .returns(Promise.resolve(subscriptionService));
+
+      const storyEl = doc.createElement('amp-story');
       subscriptionsEl = (
         <amp-story-subscriptions layout="container"> </amp-story-subscriptions>
       );
-      const storyEl = doc.createElement('amp-story');
-      storyEl.appendChild(subscriptionsEl);
-      doc.body.appendChild(storyEl);
-
       storySubscriptions = new AmpStorySubscriptions(subscriptionsEl);
+      storyEl.appendChild(subscriptionsEl);
+      win.document.body.appendChild(storyEl);
+      await subscriptionsEl.whenBuilt();
     });
 
-    it('should contain amp-subscriptions attributes', async () => {
-      await subscriptionsEl.whenBuilt();
+    it('should contain amp-subscriptions attributes: subscriptions-dialog', async () => {
       expect(
         subscriptionsEl
           .querySelector('div')
           .hasAttribute('subscriptions-dialog')
       ).to.equal(true);
+    });
+
+    it('should contain amp-subscriptions attributes: subscriptions-display', async () => {
       expect(
         subscriptionsEl
           .querySelector('div')
           .getAttribute('subscriptions-display')
       ).to.equal('NOT granted');
     });
 
-    it('should display the blocking paywall on state update', async () => {
-      await storySubscriptions.buildCallback();
-      await nextTick();
+    it('should activate subscription platform and show paywall on dialog UI state update to true', async () => {
+      const maybeRenderDialogForSelectedPlatformSpy = env.sandbox.spy(
+        subscriptionService,
+        'maybeRenderDialogForSelectedPlatform'
+      );
 
       storeService.dispatch(Action.TOGGLE_SUBSCRIPTIONS_DIALOG_UI_STATE, true);
+      await afterRenderPromise(win);
+
+      it('paywall element should have visible class', () => {
```

Comment:
Done.
 
So for the nested it, the rationale behind it is to treat the outer it a single test case. It is still one single test case but just checking two variables. So instead of setting things up in beforeEach, I actually make it global to run the setup only once. I think maybe both way works so I change it anyway. Reference from the last PR: https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/test/test-amp-story.js#L1440-L1465